### PR TITLE
Make change revision available to consumers.

### PR DESCRIPTION
**Pull request by Jamie-sas**
JIRA JENKINS-23873

This change  must be in place before the fix for the Gerrit Trigger Plugin will work.

### CODE REVIEW

**Comment by Jamie-sas** [comment]
In case someone needs a bit more info...

This change makes the merged SHA-1 available, as well as the plain patchset SHA-1 for a change that caused a merge.

We were running into problems where a Jenkins validation build of a Gerrit submit would work, but a build after the patch was merged would fail.  This was due to the patch itself being fine, but not working with another patch it merged with.

In order to combat the problem, we wanted to do the build based on the merged source, but that value was unavailable to Jenkins due to this jar not surfacing the revision... it was in the event stream from Gerrit, but the jar didn't have any code to parse if from the JSON and supply it upstream.

So, I fixed that, and made the Gerrit Trigger plugin create a new env var based on this value, named "GERRIT_MERGED_REVISION".  If no merge occurs (such as a plain fast-forward push), the value will be identical to GERRIT_PATCHSET_REVISION, but in the event of a merge, it is a different value - the SHA-1 of the merged commit.

This allows you to set up Jenkins to build the patchset itself, the merged set, or both, as needed.

File: src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/events/ChangeMerged.java [hunk 1]
```diff
@@ -80,11 +83,21 @@ public String getModifiedRef() {
         return null;
     }
 
+    /**
+     * @return ChangeMerge revision
+     */
+    public String getMergedRevision() {
```
Comment (rsandell) [inline src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/events/ChangeMerged.java]: I would prefer the property has the same name _(in camelCase)_ as the json property.

**Comment by Jamie-sas** [comment]
I thought about that, but there is already a NEWREV in the patchset, so it might be confusing as to which NEWREV to use.  Not sure why the Gerrit folks reused the same name for this case.

However, it is easy enough to change if you feel strongly about it.

**Comment by rsandell** [comment]
From what I can see `NEWREV` is only used in `RefUpdated` events and there it means the same thing as here, so it is quite fitting to name it the same, or am I missing something?

**Comment by Jamie-sas** [comment]
No, not as far as I can see.  I was just trying to be more explicit about exactly what type of 'rev' or 'ref' this is.  'new' doesn't really convey that.

Does this mean you prefer the property be named 'newRev', but the 'getMergedRevision()' getter name is OK, or you would prefer they be changed to 'newRev' and 'getNewRev()'?

**Comment by rsandell** [comment]
Should be `getNewRev()`, same as in RefUpdated, with the field having the same name so that the bean standard is upheld :)

**Comment by Jamie-sas** [comment]
Changes completed.  And 'polygon' did basically the same thing I did.

**Comment by rsandell** [comment]
@Jamie-sas The diff still looks the same was the mege applied in correct order?

**Comment by rsandell** [comment]
Merged #55 instead

**Pull request closed without merging**